### PR TITLE
Xslt support for the file configuration

### PR DESCRIPTION
**Pull request by diegolovison**
Example of the benchmark configuration
```xml
<config name="dist">
  <setup group="server" plugin="${env.PLUGINNAME}">
    <server xmlns="urn:radargun:plugins:${env.PLUGINNAME}:3.0" file="xslt(file_path, xslt_path)" jmx-domain="jboss.datagrid-infinispan" cache-manager-name="clustered" start-timeout="120000" />
  </setup>
</config>
```

### CODE REVIEW

**Comment by diegolovison** [comment]
In order to fix the comment https://github.com/radargun/radargun/pull/584#issuecomment-440381686 we introduced a third argument for the xslt function.

`xslt(file_path, xslt_path, output_path)` will create the `output.xml` into the `output_path`. The issue is that the `file_path` is into the zip file and it doesn't exist.

```xml
         <setup group="server" plugin="${env.PLUGINNAME}">
            <server xmlns="urn:radargun:plugins:${env.PLUGINNAME}:3.0" file="xslt(/Users/dlovison/Documents/RadarGunLocalRun/servers/server${slave.index}/standalone/configuration/clustered.xml, ${env.WORKSPACE}/jdg-qe/configs/${env.PLUGINNAME}-server/regression/${env.TRANSFORM_CONFIGURATION_FILE:clustered-transform.xslt}, /Users/dlovison/Documents/RadarGunLocalRun/servers/config${slave.index})">
               <args>
               </args>
               <home>/Users/dlovison/Documents/RadarGunLocalRun/servers/server${slave.index}</home>
               <server-zip>/Users/dlovison/Documents/RadarGunLocalRun/infinispan-server-9.4.0-SNAPSHOT.zip</server-zip>
            </server>
         </setup>
```

This is why we need to execute the `xslt` transformation later.

**Comment by diegolovison** [comment]
With the commit https://github.com/radargun/radargun/pull/584/commits/f23e4b7bbbb43856bdd2f8ae90a7b61397e15916 RadarGun support the `xslt` in all `ProcessService`. Also, it allows the file be evaluated later when a custom business logic is required

File: plugins/tomcat8/src/main/java/org/radargun/service/TomcatServerService.java [hunk 1]
```diff
@@ -85,31 +85,34 @@ public void init() {
...
-      try {
-         URL resource = getClass().getResource("/" + file);
-         Path filesystemFile = FileSystems.getDefault().getPath(file);
-         Path target = FileSystems.getDefault().getPath(catalinaBase, "conf", "radargun-tomcat-" + ServiceHelper.getContext().getSlaveIndex() + ".xml");
-         if (resource != null) {
-            try (InputStream is = resource.openStream()) {
-               log.info("Found " + file + " as a resource");
-               Files.copy(is, target, StandardCopyOption.REPLACE_EXISTING);
+
+      evaluateFile((evaluatedFile) -> {
```
Comment (alanfx) [inline plugins/tomcat8/src/main/java/org/radargun/service/TomcatServerService.java]: Once again, I wonder if there is a way to remove some of this boiler plate code? Maybe a method in the `ProcessService` class?

Comment (diegolovison) [reply plugins/tomcat8/src/main/java/org/radargun/service/TomcatServerService.java]: When https://github.com/radargun/radargun/pull/584/files#diff-096483ddb920163d93ac6499fb4e8320R61 is overwritten each class has a different behavior.

https://github.com/radargun/radargun/pull/584/files#diff-58c2bbab38ba23ee91c39b662dc67ec3R89 is the custom code for tomcat

https://github.com/radargun/radargun/pull/584/files#diff-3fb0eecefe0cce27cfd6a152bf6e968aR105 is the custom code for Infinispan

The boiler plate code is https://github.com/radargun/radargun/pull/584/files#diff-096483ddb920163d93ac6499fb4e8320R120

When we don't overwrite https://github.com/radargun/radargun/pull/584/files#diff-096483ddb920163d93ac6499fb4e8320R61 there is no additional code and `ProcessService` will do the transformation. See https://github.com/radargun/radargun/pull/584/files#diff-096483ddb920163d93ac6499fb4e8320R66

File: plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java [hunk 2]
```diff
@@ -100,27 +101,35 @@ public void init() {
...
-               Files.copy(is, target, StandardCopyOption.REPLACE_EXISTING);
+
+         evaluateFile((evaluatedFile) -> {
+            Path filesystemFile;
+            try {
+               URL resource = getClass().getResource("/" + evaluatedFile);
+               filesystemFile = FileSystems.getDefault().getPath(evaluatedFile);
+               Path target = FileSystems.getDefault().getPath(home, "standalone", "configuration", "radargun-" + ServiceHelper.getContext().getSlaveIndex() + ".xml");
+
+               if (resource != null) {
```
[the remark this answers is not in the document]
Comment (diegolovison) [reply plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java]: Sounds good. 
We could create `ServerService`.
`ServerService extends JavaProcessService`
`TomcatServerService extends ServerService`
`InfinispanServerService extends ServerService`
What do you think?

[the remark this answers is not in the document]
Comment (diegolovison) [reply plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java]: Done

File: core/src/test/resources/input.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<catalog>
+   <cd>
+      <title>Empire Burlesque</title>
+      <artist>Bob Dylan</artist>
```
Comment (alanfx) [inline core/src/test/resources/input.xml]: If I haven't said this already, I really appreciate your musical tastes! 💯

Comment (diegolovison) [reply core/src/test/resources/input.xml]: Well, it comes from https://www.w3schools.com/xml/xsl_transformation.asp :D

Comment (alanfx) [reply core/src/test/resources/input.xml]: Another one of my dreams is shattered! 🎼

**Pull request merged**